.github/CONTRIBUTING.markdown

```diff
@@ -1,105 +1,121 @@
+2. Click the pencil icon in the top right corner to edit the file
+3. Make your proposed changes
+4. Click "Propose file change"
+5. Click "Create pull request"
+6. Add a descriptive title and detailed description for your proposed change. The more information the better.
+7. Click "Create pull request"
+
+That's it! You'll be automatically subscribed to receive updates as others review your proposed change and provide feedback.
+
+### Submitting a pull request via Git command line
+
+1. Fork the project by clicking "Fork" in the top right corner of [`jekyll/jekyll`](https://github.com/jekyll/jekyll)
+2. Clone the repository lcoally `git clone https://github.com/<you-username>/jekyll`
+3. Create a new, descriptively named branch to contain your change ( `git checkout -b my-awesome-feature` ).
+4. Hack away, add tests. Not necessarily in that order.
+5. Make sure everything still passes by running `script/cibuild` (see [the tests section](#running-tests-locally) below)
+6. Push the branch up ( `git push origin my-awesome-feature` ).
+7. Create a pull request by visiting https://github.com/<your-username>/jekyll/ and following the instructions at the top of the screen.
+
+## Proposing updates to the documentation
+
+We want the Jekyll documentation to be the best it can be. We've open-sourced our docs and we welcome any pull requests if you find it lacking.
+
+### How to submit changes
+
+You can find the documentation for jekyllrb.com in the [site](https://github.com/jekyll/jekyll/tree/master/site) directory. See the section above, [submitting a pull request](#submitting-a-pull-request) for information on how to propose a change.
+
+One gotcha, all pull requests should be directed at the `master` branch (the default branch).
 
-Workflow
---------
+### Adding plugins
 
-Here's the most direct way to get your work merged into the project:
+If you want to add your plugin to the [list of plugins](http://jekyllrb.com/docs/plugins/#available-plugins), please submit a pull request modifying the [plugins page source file](site/_docs/plugins.md) by adding a link to your plugin under the proper subheading depending upon its type.
 
-* [Fork](https://github.com/jekyll/jekyll/fork) the project.
-* Clone down your fork ( `git clone git@github.com:[username]/jekyll.git` ).
-* Create a topic branch to contain your change ( `git checkout -b my_awesome_feature` ).
-* Hack away, add tests. Not necessarily in that order.
-* Make sure everything still passes by running `script/cibuild`.
-* If necessary, rebase your commits into logical chunks, without errors.
-* Push the branch up ( `git push origin my_awesome_feature` ).
-* Create a pull request against jekyll/jekyll and describe what your change
-  does and the why you think it should be merged.
+## Code Contributions
 
-Updating Documentation
-----------------------
+Interesting in submitting a pull request? Awesome. Read on. There's a few common gotchas, we'd love to help you avoid.
```

Comment:
I don't believe you need a comma in the sentence `There's a few common gotchas, we'd love to help you avoid`. I know of two different ways to improve this sentence.

1\. Remove the comma :smiley:

**or**

2\. Split the sentence: `There's a few common gotchas. We'd love to help you avoid them.`